### PR TITLE
chore(builder, logger, logspout): remove unused references to BUILD_IMAGE

### PR DESCRIPTION
**Pull request by ngpestelos**
...from the Makefile

### CODE REVIEW

**Comment by mboersma** [comment]
You're right, we don't actually use `$BUILD_IMAGE` in logger or builder. Code LGTM.

**Comment by carmstrong** [comment]
Code LGTM. Thanks @ngpestelos!

**Comment by bacongobbler** [comment]
We also don't use it in logspout, either. Could you please remove those references there as well? :) Thanks!

File: logspout/Makefile [hunk 1]
```diff
@@ -12,12 +11,11 @@ build: check-docker
 
 clean: check-docker check-registry
```
Comment (bacongobbler) [inline logspout/Makefile]: we no longer need these pre-trip checks for `make clean`

**Comment by mboersma** [comment]
Thanks for being so flexible @ngpestelos. Code LGTM.

**Pull request merged**